src/backend/gporca/libgpos/src/_api.cpp

```diff
@@ -280,20 +257,47 @@ gpos_exec(gpos_exec_params *params)
 //
 //---------------------------------------------------------------------------
 void
-gpos_terminate()
+gpos_terminate(bool shutdown)
 {
 #ifdef GPOS_DEBUG_COUNTERS
 	CDebugCounter::Shutdown();
 #endif
+
 #ifdef GPOS_DEBUG
+	shutdown = true;
+#endif	// GPOS_DEBUG
+
+	if (!shutdown)
+	{
+		return;
+	}
+
 #ifdef GPOS_FPSIMULATOR
-	CFSimulator::FSim()->Shutdown();
+	if (NULL != CFSimulator::FSim())
+	{
+		CFSimulator::FSim()->Shutdown();
+	}
 #endif	// GPOS_FPSIMULATOR
-	CMessageRepository::GetMessageRepository()->Shutdown();
-	CWorkerPoolManager::WorkerPoolManager()->Shutdown();
-	CCacheFactory::GetFactory()->Shutdown();
-	CMemoryPoolManager::GetMemoryPoolMgr()->Shutdown();
-#endif	// GPOS_DEBUG
+
+	if (NULL != CMessageRepository::GetMessageRepository())
```

Comment:
CMessageRepository::GetMessageRepository() always returns not NULL, because of `GPOS_ASSERT(NULL != m_repository);` in this method. If malloc returns 0 at `gpos::clib::Malloc (size=304) at clibwrapper.cpp:475`, then m_repository is NULL

Reply:
fixed